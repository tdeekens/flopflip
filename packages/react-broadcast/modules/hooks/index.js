export { default as useFeatureToggle } from './use-feature-toggle';
