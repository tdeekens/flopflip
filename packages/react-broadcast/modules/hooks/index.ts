export { default as useFeatureToggle } from './use-feature-toggle';
export { default as useAdapterStatus } from './use-adapter-status';
export {
  default as useAdapterReconfiguration,
} from './use-adapter-reconfiguration';
