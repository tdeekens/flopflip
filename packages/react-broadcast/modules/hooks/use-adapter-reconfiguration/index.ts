export { useAdapterReconfiguration as default } from '@flopflip/react';
