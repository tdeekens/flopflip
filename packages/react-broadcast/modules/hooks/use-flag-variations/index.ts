export { default } from './use-flag-variations';
