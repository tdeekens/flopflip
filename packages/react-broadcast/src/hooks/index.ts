export { useAdapterReconfiguration } from './use-adapter-reconfiguration';
export { useAdapterStatus } from './use-adapter-status';
export { useAllFeatureToggles } from './use-all-feature-toggles';
export { useFeatureToggle } from './use-feature-toggle';
export { useFeatureToggles } from './use-feature-toggles';
export { useFlagVariation } from './use-flag-variation';
export { useFlagVariations } from './use-flag-variations';
