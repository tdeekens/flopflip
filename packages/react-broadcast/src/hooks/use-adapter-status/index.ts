export { useAdapterStatus } from './use-adapter-status';
