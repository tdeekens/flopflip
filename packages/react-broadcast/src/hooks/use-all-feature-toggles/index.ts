export { useAllFeatureToggles } from './use-all-feature-toggles';
