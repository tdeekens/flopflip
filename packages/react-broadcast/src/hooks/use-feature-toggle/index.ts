export { default } from './use-feature-toggle';
