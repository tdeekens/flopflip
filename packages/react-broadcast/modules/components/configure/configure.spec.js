import React from 'react';
import { render as rtlRender, act } from '@flopflip/test-utils';
import adapter, { updateFlags } from '@flopflip/memory-adapter';
import { useFeatureToggle, useAdapterStatus } from '../../hooks';
import Configure from './configure';

const testFlagName = 'firstFlag';
const TestComponent = () => {
  const { isUnconfigured, isConfiguring, isConfigured } = useAdapterStatus();

  const isFeatureEnabled = useFeatureToggle(testFlagName);

  return (
    <ul>
      <li>Is unconfigured: {isUnconfigured ? 'Yes' : 'No'}</li>
      <li>Is configuring: {isConfiguring ? 'Yes' : 'No'}</li>
      <li>Is configured: {isConfigured ? 'Yes' : 'No'}</li>
      <li>Feature enabled: {isFeatureEnabled ? 'Yes' : 'No'}</li>
    </ul>
  );
};

const createTestProps = (custom) => ({
  adapter,
  adapterArgs: {
    fooId: 'foo-id',
  },

  ...custom,
});

const render = () => {
  const props = createTestProps();
  const rendered = rtlRender(
    <Configure {...props}>
      <TestComponent />
    </Configure>
  );
  const waitUntilConfigured = () => rendered.findByText(/Is configured: Yes/i);

  return { ...rendered, waitUntilConfigured };
};

describe('when feature is disabled', () => {
  it('should indicate the feature being disabled', async () => {
    const rendered = render();

    await rendered.waitUntilConfigured();

    expect(rendered.getByText(/Feature enabled: No/i)).toBeInTheDocument();
  });
});

describe('when enabling feature is', () => {
  it('should indicate the feature being enabled', async () => {
    const rendered = render();

    await rendered.waitUntilConfigured();

    act(() =>
      updateFlags({
        [testFlagName]: true,
      })
    );

    expect(rendered.getByText(/Feature enabled: Yes/i)).toBeInTheDocument();
  });
});

describe('when unconfigured', () => {
  it('should indicate through the adapter state', async () => {
    const rendered = render();

    expect(rendered.getByText(/Is unconfigured: Yes/i)).toBeInTheDocument();
    expect(rendered.getByText(/Is configuring: No/i)).toBeInTheDocument();
    expect(rendered.getByText(/Is configured: No/i)).toBeInTheDocument();

    await rendered.waitUntilConfigured();
  });
});

describe('when configured', () => {
  it('should indicate through the adapter state', async () => {
    const rendered = render();

    await rendered.waitUntilConfigured();

    expect(rendered.getByText(/Is configuring: No/i)).toBeInTheDocument();
    expect(rendered.getByText(/Is configured: Yes/i)).toBeInTheDocument();
  });
});

describe('statics', () => {
  describe('displayName', () => {
    it('should be set to `ConfigureFlopflip`', () => {
      expect(Configure.displayName).toEqual('ConfigureFlopflip');
    });
  });

  describe('defaultProps', () => {
    it('should default `defaultFlags` to an empty object', () => {
      expect(Configure.defaultProps.defaultFlags).toEqual({});
    });

    it('should default `shouldDeferAdapterConfiguration` to `true`', () => {
      expect(Configure.defaultProps.shouldDeferAdapterConfiguration).toBe(
        false
      );
    });
  });
});
