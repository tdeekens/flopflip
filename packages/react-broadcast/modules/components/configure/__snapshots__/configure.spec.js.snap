// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering should match snapshot 1`] = `
<ContextProvider
  value={Object {}}
>
  <ConfigureAdapter
    adapter={
      Object {
        "configure": [MockFunction],
        "getIsReady": [MockFunction],
        "reconfigure": [MockFunction],
      }
    }
    adapterArgs={
      Object {
        "fooId": "foo-id",
        "onFlagsStateChange": [Function],
        "onStatusStateChange": [Function],
      }
    }
    defaultFlags={Object {}}
    render={null}
    shouldDeferAdapterConfiguration={false}
  >
    <ChildComponent />
  </ConfigureAdapter>
</ContextProvider>
`;
